### PR TITLE
Update dependencies and fix compatibility with python3

### PR DESCRIPTION
**Pull request by tichakorn832**
WARNING: THIS IS NOT TO BE MERGED YET. I'm opening a PR to see whether I'm on the right track.

The purpose of this PR is to fix compatibility with python3 (the target is python 3.7) and update requirements and the setup script to the more up-to-date versions. During this change, many tests fail, which I'm trying to fix.

* dumpsmach_test now runs with python 3.7, provided that a minor fix to the omega package is made.

FYI: @slivingston @murrayrm

### CODE REVIEW

**Comment by murrayrm** [comment]
Is there a reason to generate the PR now rather than just testing things out in the branch and then doing the PR once things are fixed?  I don't think it really matters, but this PR will get a bit cluttered as you debug...

**Comment by tichakorn832** [comment]
I created a PR early to (hopefully) get some help to make sure that I'm on the right track. (I agree that a branch would work too but with a PR, it's easier to provide comments.)

**Comment by slivingston** [comment]
for future reference, it is possible to mark a pull request as a [draft](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests#draft-pull-requests).

@tichakorn832 is there any reason to not also support Python 3.8 ?

**Comment by slivingston** [comment]
also, the issue about supporting Python 3.7 and 3.8: #218

**Comment by tichakorn832** [comment]
Oh I didn't know about draft. Thanks for pointing out.

I just haven't tested on python3.8. There is no change that is specific to 3.7 though.

**Comment by slivingston** [comment]
@tichakorn832 let me know if I can be useful here. We are not using it in Tulip yet, but a good tool for testing against a variety of Python versions is [tox](https://tox.readthedocs.io/en/latest/) ([on PyPI](https://pypi.org/project/tox/)).

**Comment by tichakorn832** [comment]
@slivingston Thanks. it would be nice if you could help take a look at the change in tulip/interfaces/omega.py to make sure that I formulate it correctly with respect to the new interface in omega. There seems to be interface change in the latest version of omega, making the previous code in tulip not working anymore. All the tests run fine except test_spec_realizable_bool  in gridworld_test.py that keeps running out of memory.

**Comment by slivingston** [comment]
I am able to locally run test_spec_realizable_bool without error on GNU/Linux with Python 3.7.6.

@tichakorn832 Are you able to run the test locally? If so, perhaps the error arises from the configuration of Travis CI hosts. I will investigate.

File: setup.py [hunk 1]
```diff
@@ -96,10 +96,10 @@ def run_setup():
         license='BSD',
         classifiers=classifiers,
         install_requires=[
-            'networkx >= 1.8, <= 1.10',
+            'networkx >= 2.0, <= 2.4',
             'numpy >= 1.7',
-            'omega >= 0.0.9, < 0.1.0',
-            'ply >= 3.4',
+            'omega >= 0.1.2',
```
Comment (johnyf) [inline setup.py]: I suggest pinning to `omega == 0.2.0` or in a range `omega >= 0.1.2, <= 0.2.0`, to allow for changes in the API of `omega`.

Comment (slivingston) [reply setup.py]: @johnyf I agree. I am preparing several commits to add to this PR from my temporary branch `tmp-debug-travisci`.

Comment (murrayrm) [reply setup.py]: Planning to accept this change as is, and let @slivingston fix this in `tmp-debug-travisci`.

Comment (slivingston) [reply setup.py]: Confirmed. I will open a separate PR with my changes and those from the branch `completed_update_to_networkx_2` by @johnyf

**Pull request merged**